vector/src/androidTest/java/im/vector/app/ui/TrafficLightClientTest.kt

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.ui
+
+import android.Manifest
+import androidx.test.espresso.IdlingPolicies
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.GrantPermissionRule
+import im.vector.app.espresso.tools.ScreenshotFailureRule
+import im.vector.app.features.MainActivity
+import im.vector.app.ui.robot.ElementRobot
+import okhttp3.MediaType
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import org.json.JSONObject
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import java.lang.Thread.sleep
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+/**
+ * TrafficLight client is a UI test that receives requests from a TrafficLight server and fufils them on the UI
+ */
+@RunWith(AndroidJUnit4::class)
+class TrafficLightClientTest {
+
+    @get:Rule
+    val testRule: RuleChain = RuleChain
+            .outerRule(ActivityScenarioRule(MainActivity::class.java))
+            .around(GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE))
+            .around(ScreenshotFailureRule())
```

Comment:
if you're relying on the login flow you can make use of the `ClearCurrentSessionRule` https://github.com/vector-im/element-android/blob/develop/vector/src/androidTest/java/im/vector/app/CantVerifyTest.kt#L41 otherwise subsequent test runs can put the app in an already logged in state